librz/core/cmd/cmd_analysis.c

```diff
@@ -6007,6 +5535,325 @@ RZ_IPI RzCmdStatus rz_il_vm_status_handler(RzCore *core, int argc, const char **
 	return RZ_CMD_STATUS_OK;
 }
 
+static inline RzGraph *core_graph(RzCore *core, RzCoreGraphType type) {
+	RzGraph *graph = NULL;
+	switch (type) {
+	case RZ_CORE_GRAPH_TYPE_DATAREF:
+	case RZ_CORE_GRAPH_TYPE_DATAREF_GLOBAL:
+		graph = rz_core_analysis_graph_datarefs(core, type != RZ_CORE_GRAPH_TYPE_DATAREF_GLOBAL ? core->offset : UT64_MAX);
+		break;
+	case RZ_CORE_GRAPH_TYPE_FUNCALL:
+	case RZ_CORE_GRAPH_TYPE_FUNCALL_GLOBAL:
+		graph = rz_core_analysis_graph_callgraph(core, type != RZ_CORE_GRAPH_TYPE_FUNCALL_GLOBAL ? core->offset : UT64_MAX);
+		break;
+	case RZ_CORE_GRAPH_TYPE_DIFF: break;
+	case RZ_CORE_GRAPH_TYPE_BLOCK_FUN: break;
+	case RZ_CORE_GRAPH_TYPE_IMPORT:
+		graph = rz_core_analysis_graph_importxrefs(core);
+		break;
+	case RZ_CORE_GRAPH_TYPE_REF:
+	case RZ_CORE_GRAPH_TYPE_REF_GLOBAL:
+		graph = rz_core_analysis_graph_coderefs(core, type != RZ_CORE_GRAPH_TYPE_REF_GLOBAL ? core->offset : UT64_MAX);
+		break;
+	case RZ_CORE_GRAPH_TYPE_LINE: break;
+	case RZ_CORE_GRAPH_TYPE_XREF:
+		graph = rz_core_analysis_graph_codexrefs(core, core->offset);
+		break;
+	case RZ_CORE_GRAPH_TYPE_CUSTOM: break;
+	default:
+		rz_warn_if_reached();
+		break;
+	}
+	return graph;
+}
+
+static inline RzCmdStatus graph_handler(RzCore *core, RzCoreGraphType type, RzCoreGraphFormat format) {
+	RzGraph *graph = core_graph(core, type);
+	if (!graph) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	static const bool callgraphs[256] = {
+		false,
+		[RZ_CORE_GRAPH_TYPE_FUNCALL] = true,
+		[RZ_CORE_GRAPH_TYPE_FUNCALL_GLOBAL] = true,
+		[RZ_CORE_GRAPH_TYPE_REF] = true,
+		[RZ_CORE_GRAPH_TYPE_REF_GLOBAL] = true,
+		[RZ_CORE_GRAPH_TYPE_XREF] = true,
+	};
+
+	bool old_is_callgraph = core->graph->is_callgraph;
+	core->graph->is_callgraph = callgraphs[type];
+	rz_core_graph_print(core, graph, -1, true, format);
+	rz_graph_free(graph);
+	core->graph->is_callgraph = old_is_callgraph;
+	return RZ_CMD_STATUS_OK;
+}
+
+static inline RzCoreGraphFormat graph_format_from_string(const char *x) {
+	const char short_opt = (char)(x && strlen(x) == 1 ? x[0] : 0);
+
+	if (strncmp(x, "ascii", sizeof("ascii")) == 0 || RZ_STR_ISEMPTY(x) || short_opt == ' ') {
```

Comment:
why `strncmp`? I think `strcmp` makes more sense, as we don't want to recognize 'asciiXXXXXX` as `ascii` option.